<a href="https://colab.research.google.com/github/victormaciase/EconometriaFinanciera_MagisterFinanzasWK/blob/main/Lab5/Lab5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Laboratorio 5**


In [1]:
import pandas as pd
import statsmodels.api as sm
from scipy.stats import f
from statsmodels.stats.outliers_influence import variance_inflation_factor

## **Pregunta 1**

Carga el archivo Excel que contiene los datos de precios de vivienda y sus características.

In [2]:
# URL raw del archivo Excel en GitHub
url_data = "https://raw.githubusercontent.com/victormaciase/EconometriaFinanciera_MagisterFinanzasWK/refs/heads/main/Lab5/hprice1.xls"

# Definir los nombres de las columnas para archivo
column_names = ['price', 'assess', 'bdrms', 'lotsize', 'sqrft', 'colonial', 'lprice', 'lassess', 'llotsize', 'lsqrft']

# Cargar el archivo Excel en un DataFrame de pandas
# header=None para indicar que el archivo no tiene una fila de encabezado
# names=column_names para asignar los nombres definidos
# index_col=None para evitar que pandas use alguna columna como índice
df = pd.read_excel(url_data, header=None, names=column_names, index_col=None)

In [3]:
# Mostrar las primeras 5 filas del DataFrame con los nombres de columnas correctos
display(df.head())

,price,assess,bdrms,lotsize,sqrft,colonial,lprice,lassess,llotsize,lsqrft
0,300.0,349.1,4,6126,2438,1,5.703783,5.855359,8.720297,7.798934
1,370.0,351.5,3,9903,2076,1,5.913503,5.862210,9.200593,7.638198
2,191.0,217.7,3,5200,1374,0,5.252274,5.383118,8.556414,7.225482
3,195.0,231.8,3,4600,1448,1,5.273000,5.445875,8.433811,7.277938
4,373.0,319.1,4,6095,2514,1,5.921578,5.765504,8.715224,7.829630


## **Pregunta 2**

Calcula la matriz de correlación entre `lotsize` y `sqrft`. Calcula el *variance inflation factor (VIF)* ¿Qué puedes concluir acerca de la presencia de multicolinealidad?

In [16]:
df[['lotsize', 'sqrft']].corr()

,lotsize,sqrft
lotsize,1.000000,0.183842
sqrft,0.183842,1.000000


In [17]:
# Calcula el VIF para las variables lotsize y sqrft
if 'const' not in df_con_constante.columns:
    df_con_constante = sm.add_constant(df_con_constante)

vif_data = pd.DataFrame()
vif_data["feature"] = ['lotsize', 'sqrft']
vif_data["VIF"] = [variance_inflation_factor(df_con_constante[['const', 'lotsize', 'sqrft']].values, i)
                  for i in range(df_con_constante[['const', 'lotsize', 'sqrft']].shape[1])][1:] # Excluir la constante del VIF

display(vif_data)

,feature,VIF
0,lotsize,1.03498
1,sqrft,1.03498


*Nota:* El *factor inflacionario de varianza (VIF, por sus siglas en inglés)* es igual a:

$$VIF=\frac{1}{1-R^2_1}$$

donde $R^2_1$ es el $R^2$ de la regresión:

$$lotsize=\gamma_0+\gamma_1sqrft+u$$

Por lo tanto, $R^2_1=0.183842^2$ y

$$VIF=\frac{1}{1-0.183842^2}=1.03498$$

## **Pregunta 3**

Estima por MCO el siguiente modelo:

$$price=\beta_0+\beta_1lotsize+\beta_2sqrft+\epsilon$$

¿Tienen los coeficientes los signos esperados?

Usando un nivel de significancia del 5% ¿Qué variables son estadísticamente significativas? ¿Cómo cambia la significancia estadística al usar las correcciones a los errores estándar dados por `HC1`y `HC3`?

In [5]:
df_con_constante = sm.add_constant(df)
modelo1a = sm.OLS(df['price'], df_con_constante[['const', 'lotsize', 'sqrft']]).fit()
print(modelo1a.summary())

                            OLS Regression Results                            
Dep. Variable:                  price   R-squared:                       0.663
Model:                            OLS   Adj. R-squared:                  0.655
Method:                 Least Squares   F-statistic:                     83.67
Date:                Mon, 24 Aug 2026   Prob (F-statistic):           8.25e-21
Time:                        23:06:48   Log-Likelihood:                -484.10
No. Observations:                  88   AIC:                             974.2
Df Residuals:                      85   BIC:                             981.6
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          5.9324     23.512      0.252      0.8

In [6]:
# Corrección de errores estándar (HC1)
modelo1b = sm.OLS(df['price'], df_con_constante[['const', 'lotsize', 'sqrft']]).fit(cov_type = 'HC1')
print(modelo1b.summary())

                            OLS Regression Results                            
Dep. Variable:                  price   R-squared:                       0.663
Model:                            OLS   Adj. R-squared:                  0.655
Method:                 Least Squares   F-statistic:                     32.81
Date:                Mon, 24 Aug 2026   Prob (F-statistic):           2.76e-11
Time:                        23:07:00   Log-Likelihood:                -484.10
No. Observations:                  88   AIC:                             974.2
Df Residuals:                      85   BIC:                             981.6
Df Model:                           2                                         
Covariance Type:                  HC1                                         
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const          5.9324     33.671      0.176      0.8

In [8]:
# Corrección de errores estándar (HC3)
modelo1c = sm.OLS(df['price'], df_con_constante[['const', 'lotsize', 'sqrft']]).fit(cov_type = 'HC3')
print(modelo1c.summary())

                            OLS Regression Results                            
Dep. Variable:                  price   R-squared:                       0.663
Model:                            OLS   Adj. R-squared:                  0.655
Method:                 Least Squares   F-statistic:                     27.33
Date:                Mon, 24 Aug 2026   Prob (F-statistic):           6.83e-10
Time:                        23:07:09   Log-Likelihood:                -484.10
No. Observations:                  88   AIC:                             974.2
Df Residuals:                      85   BIC:                             981.6
Df Model:                           2                                         
Covariance Type:                  HC3                                         
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const          5.9324     36.985      0.160      0.8

## **Pregunta 4**

Estima por MCO el siguiente modelo:

$$log(price)=\beta_0+\beta_1lotsize+\beta_2sqrft+\epsilon$$

¿Tienen los coeficientes los signos esperados? Interpreta $\beta_1$ y $\beta_2$.

Usando un nivel de significancia del 5% ¿Qué variables son estadísticamente significativas? ¿Cómo cambia la significancia estadística al usar las correcciones a los errores estándar dados por `HC1`y `HC3`?

In [9]:
modelo2a = sm.OLS(df['lprice'], df_con_constante[['const', 'lotsize', 'sqrft']]).fit()
print(modelo2a.summary())

                            OLS Regression Results                            
Dep. Variable:                 lprice   R-squared:                       0.619
Model:                            OLS   Adj. R-squared:                  0.610
Method:                 Least Squares   F-statistic:                     68.98
Date:                Mon, 24 Aug 2026   Prob (F-statistic):           1.59e-18
Time:                        23:07:18   Log-Likelihood:                 22.976
No. Observations:                  88   AIC:                            -39.95
Df Residuals:                      85   BIC:                            -32.52
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          4.8098      0.074     65.063      0.0

In [10]:
# Corrección de errores estándar (HC1)
modelo2b = sm.OLS(df['lprice'], df_con_constante[['const', 'lotsize', 'sqrft']]).fit(cov_type = 'HC1')
print(modelo2b.summary())

                            OLS Regression Results                            
Dep. Variable:                 lprice   R-squared:                       0.619
Model:                            OLS   Adj. R-squared:                  0.610
Method:                 Least Squares   F-statistic:                     60.64
Date:                Mon, 24 Aug 2026   Prob (F-statistic):           4.33e-17
Time:                        23:07:22   Log-Likelihood:                 22.976
No. Observations:                  88   AIC:                            -39.95
Df Residuals:                      85   BIC:                            -32.52
Df Model:                           2                                         
Covariance Type:                  HC1                                         
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const          4.8098      0.080     60.019      0.0

In [11]:
# Corrección de errores estándar (HC3)
modelo2c = sm.OLS(df['lprice'], df_con_constante[['const', 'lotsize', 'sqrft']]).fit(cov_type = 'HC3')
print(modelo2c.summary())

                            OLS Regression Results                            
Dep. Variable:                 lprice   R-squared:                       0.619
Model:                            OLS   Adj. R-squared:                  0.610
Method:                 Least Squares   F-statistic:                     54.30
Date:                Mon, 24 Aug 2026   Prob (F-statistic):           6.41e-16
Time:                        23:07:25   Log-Likelihood:                 22.976
No. Observations:                  88   AIC:                            -39.95
Df Residuals:                      85   BIC:                            -32.52
Df Model:                           2                                         
Covariance Type:                  HC3                                         
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const          4.8098      0.084     57.027      0.0

## **Pregunta 5**

Estima por MCO el siguiente modelo:

$$log(price)=\beta_0+\beta_1log(lotsize)+\beta_2log(sqrft)+\beta_3colonial+\epsilon$$

¿Tienen los coeficientes los signos esperados? Interpreta $\beta_1$, $\beta_2$ y $\beta_3$. *Nota: colonial es una variable dummy.*

Usando un nivel de significancia del 5% ¿Qué variables son estadísticamente significativas? ¿Cómo cambia la significancia estadística al usar las correcciones a los errores estándar dados por HC1y HC3?

In [12]:
modelo3a = sm.OLS(df['lprice'], df_con_constante[['const', 'llotsize', 'lsqrft', 'colonial']]).fit(cov_type = 'HC1')
print(modelo3a.summary())

                            OLS Regression Results                            
Dep. Variable:                 lprice   R-squared:                       0.645
Model:                            OLS   Adj. R-squared:                  0.633
Method:                 Least Squares   F-statistic:                     37.38
Date:                Mon, 24 Aug 2026   Prob (F-statistic):           1.92e-15
Time:                        23:07:29   Log-Likelihood:                 26.159
No. Observations:                  88   AIC:                            -44.32
Df Residuals:                      84   BIC:                            -34.41
Df Model:                           3                                         
Covariance Type:                  HC1                                         
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const         -1.5892      0.697     -2.280      0.0

In [13]:
modelo3b = sm.OLS(df['lprice'], df_con_constante[['const', 'llotsize', 'lsqrft', 'colonial']]).fit()
print(modelo3b.summary())

                            OLS Regression Results                            
Dep. Variable:                 lprice   R-squared:                       0.645
Model:                            OLS   Adj. R-squared:                  0.633
Method:                 Least Squares   F-statistic:                     50.96
Date:                Mon, 24 Aug 2026   Prob (F-statistic):           7.34e-19
Time:                        23:07:32   Log-Likelihood:                 26.159
No. Observations:                  88   AIC:                            -44.32
Df Residuals:                      84   BIC:                            -34.41
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         -1.5892      0.598     -2.658      0.0

In [14]:
modelo3c = sm.OLS(df['lprice'], df_con_constante[['const', 'llotsize', 'lsqrft', 'colonial']]).fit(cov_type = 'HC3')
print(modelo3c.summary())

                            OLS Regression Results                            
Dep. Variable:                 lprice   R-squared:                       0.645
Model:                            OLS   Adj. R-squared:                  0.633
Method:                 Least Squares   F-statistic:                     33.36
Date:                Mon, 24 Aug 2026   Prob (F-statistic):           2.68e-14
Time:                        23:07:38   Log-Likelihood:                 26.159
No. Observations:                  88   AIC:                            -44.32
Df Residuals:                      84   BIC:                            -34.41
Df Model:                           3                                         
Covariance Type:                  HC3                                         
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const         -1.5892      0.742     -2.143      0.0

## **Pregunta 6**

Usando la especificación de la *Pregunta 5 (modelo3b)*, testea la siguiente hipótesis:

$H_0:\beta_1=\beta_2=\beta_3=0$

$H_1:$ al menos uno de los $\beta_j$ es distinto de cero.

Usando un nivel de significancia del 1% ¿Qué puedes concluir? Compara los valores obtenidos con el *output* de la pregunta anterior.

In [15]:
hipotesis = 'llotsize=0, lsqrft=0, colonial=0'
f_test = modelo3b.f_test(hipotesis)

print(f_test)


<F test: F=50.95775962704116, p=7.339029990160381e-19, df_denom=84, df_num=3>
